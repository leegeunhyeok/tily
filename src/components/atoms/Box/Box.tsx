import { forwardRef } from 'react';
import type * as Polymorphic from '@radix-ui/react-polymorphic';
import { styled } from '~/stitches.config';

const DEFAULT_TAG = 'div';

export type BoxProps = React.ComponentProps<typeof StyledBox>;

type BoxComponent = Polymorphic.ForwardRefComponent<
  typeof DEFAULT_TAG,
  BoxProps
>;

const StyledBox = styled(DEFAULT_TAG, {
  boxSizing: 'border-box',
});

const Box = forwardRef(function Box(
  props,
  forwardedRef
) {
  return (
    <StyledBox
      {...props}
      ref={forwardedRef}
    />
  );
}) as BoxComponent;


export default Box;
