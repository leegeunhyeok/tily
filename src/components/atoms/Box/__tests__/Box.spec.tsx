/**
 * @jest-environment jsdom
 */
import { render } from 'tests/test-utils';

import Box from '../Box';

const CHILDREN_TEXT = 'CHILDREN_TEXT';

describe('Box', () => {
  it('__snapshot__', () => {
    const { container } = render(<Box />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render children elements', () => {
    const { getByText } = render(<Box>{CHILDREN_TEXT}</Box>);
    expect(getByText(CHILDREN_TEXT)).toBeInTheDocument();
  });
});
