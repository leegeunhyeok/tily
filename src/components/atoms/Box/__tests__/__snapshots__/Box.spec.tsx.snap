// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box __snapshot__ 1`] = `
<div
  class="c-lesPJm"
/>
`;
